regression from 269 to 270: noqa expected on wrong line for B024

```
from abc import ABC
from dataclasses import dataclass


@dataclass
class Foo(ABC):  # noqa: B024
    """ Foo """
```

ruff check passes on v0.0.269 and fails on v0.0.270 with:

```
file.py:5:1: B024 `Foo` is an abstract base class, but it has no abstract methods
```

If the noqa is moved up to the line with @dataclass, it passes.

--- a/crates/ruff/resources/test/fixtures/flake8_bugbear/B027.py
+++ b/crates/ruff/resources/test/fixtures/flake8_bugbear/B027.py
@@ -120,3 +120,11 @@ class AbstractClass(ABC):
     @abstractmethod
     def empty_1(self, foo: Union[str, int, list, float]):
         ...
+
+
+from dataclasses import dataclass
+
+
+@dataclass
+class Foo(ABC):  # noqa: B024
+    ...

--- a/crates/ruff/src/rules/flake8_bugbear/rules/abstract_base_class.rs
+++ b/crates/ruff/src/rules/flake8_bugbear/rules/abstract_base_class.rs
@@ -2,6 +2,7 @@ use rustpython_parser::ast::{self, Constant, Expr, Keyword, Ranged, Stmt};
 
 use ruff_diagnostics::{Diagnostic, Violation};
 use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::helpers::identifier_range;
 use ruff_python_semantic::analyze::visibility::{is_abstract, is_overload};
 use ruff_python_semantic::model::SemanticModel;
 
@@ -133,7 +134,7 @@ pub(crate) fn abstract_base_class(
                 AbstractBaseClassWithoutAbstractMethod {
                     name: name.to_string(),
                 },
-                stmt.range(),
+                identifier_range(stmt, checker.locator),
             ));
         }
     }

